### ISSUE TITLE
21.24.0 breaks execution environment builds

### ISSUE THREAD

**Issue by kbreit** [ansible-collections/netapp.ontap#109]
### Summary

I use the netapp.ontap collection in execution environments. The build works fine when I use 21.23.0 but 21.24.0 introduces a change which is causing the build to error.

### Component Name

netapp.ontap

### Ansible Version

```console
(ansible-netapp) *[main][~/Projects/ansible-execution-environments/ansible-ee-netapp]$ ansible-builder --vers
ion
1.1.0
(ansible-netapp) *[main][~/Projects/ansible-execution-environments/ansible-ee-netapp]$ ansible --version
ansible [core 2.13.1]
  config file = /Users/kbreit/Projects/ansible-execution-environments/ansible-ee-netapp/ansible.cfg
  configured module search path = ['/Users/kbreit/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /Users/kbreit/.pyenv/versions/3.10.2/envs/ansible-netapp/lib/python3.10/site-packages/ansible
  ansible collection location = /Users/kbreit/.ansible/collections:/usr/share/ansible/collections
  executable location = /Users/kbreit/.pyenv/versions/ansible-netapp/bin/ansible
  python version = 3.10.2 (main, Mar 15 2022, 13:37:40) [Clang 12.0.5 (clang-1205.0.22.9)]
  jinja version = 3.1.2
  libyaml = True
```

### ONTAP Collection Version

```console
collections:
- name: netapp.ontap
  version: 21.23.0
- name: community.general
  version: 5.3.0
- name: ansible.utils
  version: 2.6.1
```

### ONTAP Version

```console
n/a
```

### Playbook

```console
---
version: 1

ansible_config: 'ansible.cfg'

build_arg_defaults:
  EE_BASE_IMAGE: 'registry.redhat.io/ansible-automation-platform-22/ee-minimal-rhel8:1.0.0-179'

dependencies:
  galaxy: requirements.yaml
  python: requirements.txt
```

### Steps to Reproduce

```yaml (paste below)
ansible-builder build --container-runtime=docker --tag myregistry.azurecr.io/ansible-ee-netapp:0.0.6 -f execution-environment.yaml
```

### Expected Results

I expected it to build with output similar to:

```
(ansible-netapp) *[main][~/Projects/ansible-execution-environments/ansible-ee-netapp]$ ansible-builder build --container-runtime=docker --tag myregistry.azurecr.io/ansible-ee-netapp:0.0.6 -f execution-environment.yaml
File context/_build/requirements.yml had modifications and will be rewritten
Running command:
  docker build -f context/Dockerfile -t myregistry.azurecr.io/ansible-ee-netapp:0.0.6 context
Complete! The build context can be found at: /Users/kbreit/Projects/ansible-execution-environments/ansible-ee-netapp/context
```

### Actual Results

```console
#17 0.768 FileNotFoundError: [Errno 2] No such file or directory: '/usr/share/ansible/collections/ansible_collections/netapp/ontap/../requirements.txt'
#17 0.768 Expected requirements file not present at: /usr/share/ansible/collections/ansible_collections/netapp/requirements.txt
#17 ERROR: executor failed running [/bin/sh -c ansible-builder introspect --sanitize --user-pip=requirements.txt --write-bindep=/tmp/src/bindep.txt --write-pip=/tmp/src/requirements.txt]: exit code: 1
------
 > [builder 4/5] RUN ansible-builder introspect --sanitize --user-pip=requirements.txt --write-bindep=/tmp/src/bindep.txt --write-pip=/tmp/src/requirements.txt:
------
executor failed running [/bin/sh -c ansible-builder introspect --sanitize --user-pip=requirements.txt --write-bindep=/tmp/src/bindep.txt --write-pip=/tmp/src/requirements.txt]: exit code: 1

An error occured (rc=1), see output line(s) above for details.
```

**Comment by lonico** [comment]
It works for me, but I'm using quay.io
```
STEP 1: FROM quay.io/ansible/ansible-runner:latest AS galaxy
...
STEP 6: FROM quay.io/ansible/ansible-builder:latest AS builder
STEP 7: ADD _build/requirements.txt requirements.txt
--> Using cache 8d8c6322db0a778c1c66780513cc3c2b21ef6e032cb2031610a93b402569d54e
--> 8d8c6322db0
STEP 8: RUN ansible-builder introspect --sanitize --user-pip=requirements.txt --write-bindep=/tmp/src/bindep.txt --write-pip=/tmp/src/requirements.txt
--> Using cache c03f0809d73ca39ebbf2cfd249a41cc49b32293305f52012b9eaa35e46f0004c
--> c03f0809d73
```

This is strange, because we did not change the location of the file.   
And everything we publish is under `ansible_collections/netapp/ontap`.   We can't have something under `ansible_collections/netapp` as it would collide with other collections.

**Comment by lonico** [comment]
I just rebuilt from scratch.   No issue using
```
dependencies:
  python: requirements.txt
```

I don't understand the .. in `No such file or directory: '/usr/share/ansible/collections/ansible_collections/netapp/ontap/../requirements.txt'` but could you try with

```
dependencies:
  python: ontap/requirements.txt
```

**Comment by kbreit-insight** [comment]
I'm seeing the same problem when using the quay.io image.

```
---
version: 1

ansible_config: 'ansible.cfg'

build_arg_defaults:
  EE_BASE_IMAGE: 'quay.io/ansible/ansible-runner:latest'

dependencies:
  galaxy: requirements.yaml
  python: requirements.txt
```

If I request `21.24.0` I am reproducing the problem.

**Comment by kbreit-insight** [comment]
I meant to add that when I try to add `ontap/requirements.txt` to the Python requirements it says it can't find that file.

**Comment by joshedmonds** [comment]
@lonico - I can confirm I'm seeing the same behaviour: https://github.com/joshedmonds/ansible-ee-builds/actions/runs/3193837950/jobs/5212804559

It also occurs if specifying the latest netapp.cloudmanager collection version.

As opposed to reverting back to the prior versions of both:
https://github.com/joshedmonds/ansible-ee-builds/actions/runs/3193887496

**Comment by lonico** [comment]
Red Hat actually asked us to add this file, and we had some internal debate on the `..`, as some stuff worked without it, and other stuff requires it.  It was our understanding this would only be used for some sanity tests.

What happens if you delete the meta/execution-environment.yml file?

**Comment by lonico** [comment]
Thank you Josh, that helped.  I can reproduce the issue.  We'll need to talk t Red Hat.

**Comment by lonico** [comment]
I confirmed that if I delete the meta/execution-environment.yml file by adding
```
RUN rm /usr/share/ansible/collections/ansible_collections/netapp/ontap/meta/execution-environment.yml
```
after the COPY of the collection
in the context/Containerfile and then run podman manually
```
sudo podman build -f context/Containerfile -t myregistry.azurecr.io/ansible-ee-netapp:0.0.6 context
```
it works.

**Comment by lonico** [comment]
similar success with
```
RUN sed -i 's|\.\.\/||' /usr/share/ansible/collections/ansible_collections/netapp/ontap/meta/execution-environment.yml
```
to remove the ../

**Comment by lonico** [comment]
The issue can be seen locally with
```
$ ansible-builder introspect --sanitize .
Expected requirements file not present at: /home/laurentn/atelier/wsl/ansible/ansible_collections/ansible_collection_ontap/ansible_collections/netapp/requirements.txt
FileNotFoundError: [Errno 2] No such file or directory: './ansible_collections/netapp/ontap/../requirements.txt'

**Comment by lonico** [comment]
As a work-around, you can pull from GitHub as
```
  # Install a collection from GitHub.
  - source: https://github.com/ansible-collections/netapp.ontap.git
    type: git
    version: test_ee_24
```
test_ee_24 is the same as 21.24.0 with a fix with the EE issue.  We will delete it once 21.24.1 is released.

**Comment by lonico** [comment]
Pushed to Galaxy as 21.24.1.  Verified with
```
---
collections:
  # Install collections from Galaxy
  # - name: ansible.posix
  # - name: netapp.aws
  # # name: - netapp.azure
  # - name: netapp.cloudmanager
  #   version: 21.19.0
  # - name: netapp.elementsw
  - name: netapp.ontap
    version: 21.24.1
  # - name: netapp.storagegrid
  # - name: netapp.um_info
```

thanks to @kbreit and @joshedmonds 

(We'll fix the same issue with CloudManager)

**Comment by lonico** [comment]
Cloud Manager was fixed in 21.20.1.
